<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>SDSPlayerClass</class>
 <widget class="QMainWindow" name="SDSPlayerClass">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>790</width>
    <height>678</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>SDSPlayer</string>
  </property>
  <property name="styleSheet">
   <string/>
  </property>
  <property name="toolButtonStyle">
   <enum>Qt::ToolButtonTextBesideIcon</enum>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout_2">
    <property name="spacing">
     <number>2</number>
    </property>
    <property name="topMargin">
     <number>2</number>
    </property>
    <property name="bottomMargin">
     <number>2</number>
    </property>
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_8">
      <property name="topMargin">
       <number>5</number>
      </property>
      <item>
       <widget class="Viewer" name="viewer" native="true">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QScrollArea" name="scrollArea">
        <property name="widgetResizable">
         <bool>true</bool>
        </property>
        <widget class="QWidget" name="scrollAreaWidgetContents">
         <property name="geometry">
          <rect>
           <x>0</x>
           <y>0</y>
           <width>380</width>
           <height>523</height>
          </rect>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_3">
          <item>
           <widget class="QTextEdit" name="textEditConsole">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_5">
      <item>
       <widget class="QComboBox" name="comboBoxViewMode">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="toolTip">
         <string>View mode</string>
        </property>
        <property name="currentIndex">
         <number>1</number>
        </property>
        <item>
         <property name="text">
          <string>wireframe</string>
         </property>
         <property name="icon">
          <iconset resource="res.qrc">
           <normaloff>:/icons/icons/wireframe.png</normaloff>:/icons/icons/wireframe.png</iconset>
         </property>
        </item>
        <item>
         <property name="text">
          <string>surface</string>
         </property>
         <property name="icon">
          <iconset resource="res.qrc">
           <normaloff>:/icons/icons/surface.png</normaloff>:/icons/icons/surface.png</iconset>
         </property>
        </item>
        <item>
         <property name="text">
          <string>smooth</string>
         </property>
         <property name="icon">
          <iconset resource="res.qrc">
           <normaloff>:/icons/icons/smooth.png</normaloff>:/icons/icons/smooth.png</iconset>
         </property>
        </item>
        <item>
         <property name="text">
          <string>cells</string>
         </property>
         <property name="icon">
          <iconset resource="res.qrc">
           <normaloff>:/icons/icons/cells.png</normaloff>:/icons/icons/cells.png</iconset>
         </property>
        </item>
       </widget>
      </item>
      <item>
       <widget class="QComboBox" name="comboBoxEditMode">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="toolTip">
         <string>Interaction mode</string>
        </property>
        <item>
         <property name="text">
          <string>view</string>
         </property>
         <property name="icon">
          <iconset resource="res.qrc">
           <normaloff>:/icons/icons/eye.png</normaloff>:/icons/icons/eye.png</iconset>
         </property>
        </item>
        <item>
         <property name="text">
          <string>inspect</string>
         </property>
         <property name="icon">
          <iconset resource="res.qrc">
           <normaloff>:/icons/icons/pencil.png</normaloff>:/icons/icons/pencil.png</iconset>
         </property>
        </item>
       </widget>
      </item>
      <item>
       <widget class="QToolButton" name="toolButtonFitToWindow">
        <property name="toolTip">
         <string>Fit to window</string>
        </property>
        <property name="text">
         <string>...</string>
        </property>
        <property name="icon">
         <iconset resource="res.qrc">
          <normaloff>:/icons/icons/magnifier.png</normaloff>:/icons/icons/magnifier.png</iconset>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QComboBox" name="comboBoxSelectMode">
        <item>
         <property name="text">
          <string>cells</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>edges</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>tetrahedra</string>
         </property>
        </item>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="pushButtonTetTopograph">
        <property name="text">
         <string>tet topograph</string>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer_2">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_6">
      <property name="topMargin">
       <number>4</number>
      </property>
      <property name="bottomMargin">
       <number>4</number>
      </property>
      <item>
       <widget class="QLineEdit" name="lineEdit">
        <property name="text">
         <string>shift-click to select an element</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout">
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_2">
        <property name="spacing">
         <number>0</number>
        </property>
        <item>
         <layout class="QVBoxLayout" name="verticalLayout">
          <property name="spacing">
           <number>1</number>
          </property>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_4">
            <item>
             <widget class="QToolButton" name="toolButtonRewind">
              <property name="toolTip">
               <string>Rewind</string>
              </property>
              <property name="text">
               <string>...</string>
              </property>
              <property name="icon">
               <iconset resource="res.qrc">
                <normaloff>:/icons/icons/control_start.png</normaloff>:/icons/icons/control_start.png</iconset>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QToolButton" name="toolButtonStop">
              <property name="toolTip">
               <string>Stop</string>
              </property>
              <property name="text">
               <string>...</string>
              </property>
              <property name="icon">
               <iconset resource="res.qrc">
                <normaloff>:/icons/icons/control_stop.png</normaloff>:/icons/icons/control_stop.png</iconset>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QToolButton" name="toolButtonPlay">
              <property name="toolTip">
               <string>Play</string>
              </property>
              <property name="text">
               <string>...</string>
              </property>
              <property name="icon">
               <iconset resource="res.qrc">
                <normaloff>:/icons/icons/control_play.png</normaloff>:/icons/icons/control_play.png</iconset>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <widget class="QSlider" name="sliderPlaybackRate">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="minimumSize">
             <size>
              <width>0</width>
              <height>10</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>16777215</width>
              <height>15</height>
             </size>
            </property>
            <property name="toolTip">
             <string>Playback rate</string>
            </property>
            <property name="value">
             <number>50</number>
            </property>
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="invertedAppearance">
             <bool>false</bool>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="TimeLineWidget" name="timeline" native="true">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="toolTip">
           <string>Timeline</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_3">
      <property name="spacing">
       <number>8</number>
      </property>
      <item>
       <widget class="QLabel" name="labelFilename">
        <property name="maximumSize">
         <size>
          <width>16777215</width>
          <height>20</height>
         </size>
        </property>
        <property name="font">
         <font>
          <pointsize>7</pointsize>
         </font>
        </property>
        <property name="toolTip">
         <string>Filename</string>
        </property>
        <property name="text">
         <string>filename</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="labelVersion">
        <property name="maximumSize">
         <size>
          <width>16777215</width>
          <height>20</height>
         </size>
        </property>
        <property name="font">
         <font>
          <pointsize>7</pointsize>
         </font>
        </property>
        <property name="toolTip">
         <string>Version/format of the file.</string>
        </property>
        <property name="text">
         <string>version</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="labelDate">
        <property name="maximumSize">
         <size>
          <width>16777215</width>
          <height>20</height>
         </size>
        </property>
        <property name="font">
         <font>
          <pointsize>7</pointsize>
         </font>
        </property>
        <property name="toolTip">
         <string>Date</string>
        </property>
        <property name="text">
         <string>date</string>
        </property>
        <property name="wordWrap">
         <bool>false</bool>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>10</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>790</width>
     <height>22</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="action_Open"/>
    <addaction name="actionQuit"/>
   </widget>
   <addaction name="menuFile"/>
  </widget>
  <action name="action_Open">
   <property name="text">
    <string>&amp;Open...</string>
   </property>
  </action>
  <action name="actionQuit">
   <property name="text">
    <string>&amp;Quit</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>TimeLineWidget</class>
   <extends>QWidget</extends>
   <header>timelinewidget.h</header>
   <container>1</container>
   <slots>
    <slot>play()</slot>
    <slot>stop()</slot>
    <slot>rewind()</slot>
   </slots>
  </customwidget>
  <customwidget>
   <class>Viewer</class>
   <extends>QWidget</extends>
   <header>viewer.h</header>
   <container>1</container>
   <slots>
    <slot>setViewMode(QString)</slot>
    <slot>showEntireScene()</slot>
   </slots>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="res.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>toolButtonPlay</sender>
   <signal>clicked()</signal>
   <receiver>timeline</receiver>
   <slot>play()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>93</x>
     <y>639</y>
    </hint>
    <hint type="destinationlabel">
     <x>106</x>
     <y>657</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>toolButtonStop</sender>
   <signal>clicked()</signal>
   <receiver>timeline</receiver>
   <slot>stop()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>64</x>
     <y>639</y>
    </hint>
    <hint type="destinationlabel">
     <x>256</x>
     <y>657</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>toolButtonRewind</sender>
   <signal>clicked()</signal>
   <receiver>timeline</receiver>
   <slot>rewind()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>35</x>
     <y>639</y>
    </hint>
    <hint type="destinationlabel">
     <x>142</x>
     <y>654</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>sliderPlaybackRate</sender>
   <signal>valueChanged(int)</signal>
   <receiver>SDSPlayerClass</receiver>
   <slot>playbackRateChanged(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>48</x>
     <y>652</y>
    </hint>
    <hint type="destinationlabel">
     <x>3</x>
     <y>636</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>comboBoxViewMode</sender>
   <signal>activated(QString)</signal>
   <receiver>viewer</receiver>
   <slot>setViewMode(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>17</x>
     <y>595</y>
    </hint>
    <hint type="destinationlabel">
     <x>65</x>
     <y>372</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>toolButtonFitToWindow</sender>
   <signal>clicked()</signal>
   <receiver>viewer</receiver>
   <slot>showEntireScene()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>208</x>
     <y>600</y>
    </hint>
    <hint type="destinationlabel">
     <x>205</x>
     <y>489</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>comboBoxSelectMode</sender>
   <signal>activated(QString)</signal>
   <receiver>SDSPlayerClass</receiver>
   <slot>selectMode(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>282</x>
     <y>601</y>
    </hint>
    <hint type="destinationlabel">
     <x>343</x>
     <y>600</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>playbackRateChanged(int)</slot>
  <slot>selectMode(QString)</slot>
 </slots>
</ui>
